feature_engine/encoding/ordinal.py

```diff
@@ -142,6 +143,7 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
             Otherwise, y needs to be passed when fitting the transformer.
         """
 
+        X, y = _check_X_y_pd_np_mismatch(X, y)
```

Comment:
I would restruture the logic here a bit:
if encoding_method is ordered, use check_x_y, otherwise use _is_dataframe()